### ISSUE TITLE
hw04 ready for grading

### ISSUE THREAD

**Issue by Tangjiahui26** [Tangjiahui26/STAT545-hw-Tang-Jiahui#6]
@vincenzocoia @gvdr @ksedivyhaley @JoeyBernhardt @mynamedaike @pgonzaleze @derekcho

SHA: commit df28240a6ade28ca8844a079919b3cdea54dad2e
[Homework04 folder](https://github.com/Tangjiahui26/STAT545-hw-Tang-Jiahui/tree/master/hw04)
[hw04-Tang-Jiahui.md](https://github.com/Tangjiahui26/STAT545-hw-Tang-Jiahui/blob/master/hw04/hw04-Tang-Jiahui.md)
[Cheatsheet-tidyr.md](https://github.com/Tangjiahui26/STAT545-hw-Tang-Jiahui/blob/master/hw04/Cheatsheet-tidyr.md)

*Thanks~*

**Comment by sevenelevenkevin** [comment]
**Overall:**
-great work!
-progress report given
-resources cited
-required number of tasks completed
-code was clear and easy to read 
-use of easyggplot2 was neat, I will definitely look into this
-detailed description of exploring the join functions
-detailed description of newly created data frame in part 2 of the hw

**Overall Suggestions:**
-maybe just use head() when printing long tables, or only show the parts you want the reader to see
-use warning=FALSE and message=FALSE so warnings and messages don't show up in your rendered md

**Suggestions on Specific Sections:**
General data reshaping and relationship to aggregation:
-activity 1: for your cheat sheet, maybe include a little more explanation on what each function is doing
-activity 2: your plot was awesome but for me I just used the function pair() as we only needed to plot lifeExpectancy against each country here, much simpler  

Join, merge, look up:
-great job here, very detailed, but you could clean a few things up, see overall suggestions  

---
Peer Reviewed by Kevin Zhu

**Comment by farihakhan** [comment]
I really liked you readme doc. It shows a nice summary of what your repo contains and which activities you did. I like that you included embedded links to the assignment outline and resources, but it would have been nice to have a link to your assignment rmd also. 

Your cheatsheet seems quite thorough and covers the use of a variety of different functions. I think using a little bit more description and commenting within your code chunks may help make the cheatsheet more useful and give it more context.

Your activity 2 was done quite well. I really like your display of graphs, the colours and the arrangement look really good. You can try playing with different themes, I find that there are some continuous themes which use colours that are a little bit more unambiguous.

The joining activity you chose was done quite thoroughly as well. Nice job with using the countrycode() function, I think using a built in function like this definitely decreases the chances of human errors. In this section, maybe using the head() function for the kable would have been a better option, to limit the size of the table that's being displayed.

Overall I think you did a great job!

**Comment by pgonzaleze** [comment]
Hi @Tangjiahui26
General data reshaping and relationship to aggregation: Yes (1 and 2)
Join, merge, look up: Yes (1)
Progress report: Yes
Extras (optional; merge/match): No
+ Your assignment is complete,
#### Data Reshaping
+ It would be nice to have more explanations of what you are doing or planning to do,
+ The coding style is neat,
+ Nice integration of functions learned, 
+ I would recommend you that when assign names try to use descriptive names rather than only a “code name” like A1, this could be very confusing for the readers, 
+ I liked the graphs, but would be amazing if you make a brief description/discussion about them

#### Join functions
+ Nice use of functions,
+ I would appreciate more comments in order to make easier the understanding, for example; mention the names of the variables that appears/disappears,
+ Same comment about the names, try to give different names otherwise it is hard to follow, E.g. You mentioned: “We can find that left_join1 will return all rows from second_gapminder and all rowa from second_gapminder and first_gapminder. Left_join2 is the opposite of left_join1”, so the reader have to go up and down and decipher what is the difference between both tables,
+ The same for names like gapminder1 and gapminder2, the reader have to go up and down to remember which is which,  
+ Nice practice,

####
+ Good report and links,
+ Good exploration and approach, 

Your marks will be distributed later,

Regards,

Pedro G
